gcloud/_helpers.py

```diff
@@ -302,14 +303,30 @@ def _total_seconds(offset):
 def _rfc3339_to_datetime(dt_str):
     """Convert a string to a native timestamp.
 
+    .. note::
+
+       Python datetimes do not support nanosecond precision;  this function
+       therefore truncates such values to microseconds.
+
     :type dt_str: str
     :param dt_str: The string to convert.
 
     :rtype: :class:`datetime.datetime`
     :returns: The datetime object created from the string.
     """
-    return datetime.datetime.strptime(
-        dt_str, _RFC3339_MICROS).replace(tzinfo=UTC)
+    try:
+        return datetime.datetime.strptime(
+            dt_str, _RFC3339_MICROS).replace(tzinfo=UTC)
+    except ValueError:  # maybe nanosecond resoultion
+        prefix, suffix = dt_str.split('.')
+        fraction, zulu = suffix[:-1], suffix[-1]
```

Comment:
I'm not trying to support more formats just trying to avoid lots of extra work / error handling.

If we always end in `Z` let's just check and strip outright or give a nice error if it is missing. We can just remove the `Z` from the format string.

Here's another question, do nanoseconds have to be exactly length 9? Do we have a guarantee?

To use native when possible why don't we locate the `.` and then special case if there are more than 6 characters after it.